ecc/secp256k1/ecdsa/ecdsa.go

```diff
@@ -185,34 +229,41 @@ func (privKey *PrivateKey) Public() signature.PublicKey {
 	return &pub
 }
 
-// Sign performs the ECDSA signature
+// SignForRecover performs the ECDSA signature and returns public key recovery information
 //
 // k ← 𝔽r (random)
 // P = k ⋅ g1Gen
 // r = x_P (mod order)
 // s = k⁻¹ . (m + sk ⋅ r)
-// signature = {r, s}
+// v = (div(x_P, order)<<1) || y_P[-1]
```

Comment:
which standard this follows? Ethereum uses 27/28 and SEC 1 seems to omit this check and call it vacuous in sec 4.1.3

Reply:
I tried to generalize for all the curve. I recall a NIST standard, will look for it.

Reply:
So, I looked here https://www.secg.org/sec1-v2.pdf Section 4.1.6. There is a loop j=0,...,h until x==r+jn. For determinstic recovery we store such j and high bit of y concatenated. Imo I didn't find this explicitly written anywhere, but it seems at least that this is the logic of constructing v.

The +27 thing in EVM is something very Ethereum-specific and I do not know where it comes from. At least for me it makes sense to have v \in [0, N] instead of [27, ..] and then if we use the public key recover in Ethereum context, just subtract 27.

Reply:
for h=1, 2 sec1-v2 omits this check and calls it vacuous at the end of page 45. But anyway it doesn't hurt to keep this + it should make sense when/if we add this methods to other curves with big cofactor.

Reply:
Kept for now.